onnxruntime/test/optimizer/graph_transform_utils_test.cc

```diff
@@ -0,0 +1,119 @@
+#include "core/framework/ml_value.h"
+#include "core/util/math.h"
+#include "core/platform/env.h"
+#include "test/framework/test_utils.h"
+#include "test/capturing_sink.h"
+#include "test/test_environment.h"
+#include "gtest/gtest.h"
+#include "core/optimizer/rule_based_graph_transformer.h"
+#include "core/optimizer/constant_folding.h"
+#include "core/optimizer/graph_transformer_utils.h"
+
+using namespace std;
+using namespace ONNX_NAMESPACE;
+
+namespace onnxruntime {
+namespace test {
+
+TEST(GraphTransformerUtilsTests, TestValidateLevels) {  
+  
+  ASSERT_TRUE(transformerutils::ValidateTransformerLevel(0).IsOK());
+  
+  ASSERT_TRUE(transformerutils::ValidateTransformerLevel(1).IsOK());
+
+  ASSERT_TRUE(transformerutils::ValidateTransformerLevel(2).IsOK());
+
+  ASSERT_FALSE(transformerutils::ValidateTransformerLevel(3).IsOK());  
+}
+
+TEST(GraphTransformerUtilsTests, TestSetTransformerContext) {
+  uint32_t levels_enabled;
+
+  transformerutils::SetTransformerContext(0, levels_enabled);
+  ASSERT_TRUE(levels_enabled == 1);
+
+  transformerutils::SetTransformerContext(1, levels_enabled);
+  ASSERT_TRUE(levels_enabled == 3);
+
+  transformerutils::SetTransformerContext(2, levels_enabled);
+  ASSERT_TRUE(levels_enabled == 7);  
+}
+
+TEST(GraphTransformerUtilsTests, TestGenerateRewriterules) {
+
+  // Generate all test
+  auto rewrite_rules = transformerutils::GenerateRewriteRules(TransformerLevel::Level1);
+  ASSERT_TRUE(rewrite_rules.size() != 0);  
+
+  // Rule name match test
+  std::vector<std::string> custom_list = {"EliminateIdentity"};
+  rewrite_rules = transformerutils::GenerateRewriteRules(TransformerLevel::Level1, &custom_list);
+  ASSERT_TRUE(rewrite_rules.size() == custom_list.size());
+
+  // 1 Rule name match test
+  custom_list = {"EliminateIdentity", "ConvAddFusion", "ConvMulFusion", "abc", "def"};
+  rewrite_rules = transformerutils::GenerateRewriteRules(TransformerLevel::Level1, &custom_list);
+  ASSERT_TRUE(rewrite_rules.size() == 1);
+
+  // Rule name no match test
+  custom_list = {"abc"};
+  rewrite_rules = transformerutils::GenerateRewriteRules(TransformerLevel::Level1, &custom_list);
+  ASSERT_TRUE(rewrite_rules.size() == 0);
+}
+
+TEST(GraphTransformerUtilsTests, TestGenerateCustomRewriterules) {
+  // Rule name match test
+  std::vector<std::string> custom_list = {"EliminateIdentity"};
+  auto rewrite_rules = transformerutils::GenerateRewriteRules(TransformerLevel::Level1, &custom_list);
+  ASSERT_TRUE(rewrite_rules.size() == custom_list.size());
+
+  // 1 Rule name match test
+  custom_list = {"EliminateIdentity", "ConvAddFusion", "ConvMulFusion", "abc", "def"};
+  rewrite_rules = transformerutils::GenerateRewriteRules(TransformerLevel::Level1, &custom_list);
+  ASSERT_TRUE(rewrite_rules.size() == 1);
+
+  // Rule name no match test
+  custom_list = {"abc"};
+  rewrite_rules = transformerutils::GenerateRewriteRules(TransformerLevel::Level1, &custom_list);
+  ASSERT_TRUE(rewrite_rules.size() == 0);
```

Comment:
How does this differ from the above test?

Reply:
This verifies when no transformer matches the method will return empty container.